### ISSUE TITLE
error for recursion limit should give instructions for increasing it

### ISSUE THREAD

**Issue by durka** [rust-lang/rust#34565]
I thought it used to do this, but I could be making it up.

**Comment by nagisa** [comment]
We still do, however there many different recursion limits. I’m not sure setting #[recursion_limit="256"] would do anything for, say, trait instantiation recursion thing.

**Comment by durka** [comment]
It doesn't (anymore?) if you exceed the limit during macro expansion.

**Comment by durka** [comment]
I'm talking about [this one](https://github.com/rust-lang/rust/blob/366de839ae9794411419c5b579c829e18adde613/src/libsyntax/ext/base.rs#L691) to be clear.

**Comment by nagisa** [comment]
IMO the optimal fix would be to introduce a helper function or macro in a form of

```
check_recursion_limit!(..., current_depth, error_msg, fmt_args)
```

and use it everywhere where we currently emit recursion error.

**Comment by durka** [comment]
Fixed by #39655 (re: https://github.com/rust-lang/rust/issues/34565#issuecomment-229620299, see #40198).
